internal/builders/bazel/README.md

```diff
@@ -0,0 +1,253 @@
+generate provenance attestations which will be uploaded as another .zip to the
+workflow run.
+
+```yaml
+jobs:
+  build:
+    permissions:
+      id-token: write # For signing
+      contents: read # For repo checkout.
+      actions: read # For getting workflow run info.
+    if: startsWith(github.ref, 'refs/tags/')
+    uses: slsa-framework/slsa-github-generator/.github/workflows/builder_bazel_slsa3.yml@v1.7.0
+    with:
+      targets: "//src:fib //src:hello"
+      flags: "--strip=always"
+```
+
+The `targets` are a set of space separated build targets to be built. The targets are assume to
+be referenced from the root of the directory. Because of this the each target inputted should be referenced
+as `//path/from/root/to/target:your_target`.
+
+Once the targets are built, the Bazel builder creates a folder for the artifacts
+and another for the provenance attestations which are uploaded as artifacts to the workflow run.
+
+### Referencing the Bazel builder
+
+At present, the builder **MUST** be referenced by a tag of the form `@vX.Y.Z`,
+because the build will fail if you reference it via a shorter tag like `@vX.Y`
+or `@vX` or if you reference it by a hash.
+
+For more information about this design decision and how to configure
+renovatebot, see the main repository [README.md](../../../README.md).
+
+### Private Repositories
+
+Private repositories are supported with some caveats. Currently all builds
+generate and post a new entry in the public
+[Rekor](https://github.com/sigstore/rekor) API server instance at
+https://rekor.sigstore.dev/. This entry includes the repository name. This will cause the
+private repository name to leak and be discoverable via the public Rekor API
+server.
+
+If this is ok with you, you can set the `rekor-log-public` flag in order to
+opt in to publishing to the public Rekor instance from a private repository.
+
+```yaml
+with:
+  rekor-log-public: true
+```
+
+If you do not set this flag then private repositories will generate an error in
+order to prevent leaking repository name information.
+
+Support for private transparency log instances that would not leak repository
+name information is tracked on [issue #372](https://github.com/slsa-framework/slsa-github-generator/issues/372).
+
+### Supported Triggers
+
+The following [GitHub trigger
+events](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows)
+are fully supported and tested:
+
+- `schedule`
+- `push` (including new tags)
+- `release`
+- Manual run via `workflow_dispatch`
+
+However, in practice, most triggers should work with the exception of
+`pull_request`. If you would like support for `pull_request`, please tell us
+about your use case on
+[issue #358](https://github.com/slsa-framework/slsa-github-generator/issues/358). If
+you have an issue in all other triggers please submit a
+[new issue](https://github.com/slsa-framework/slsa-github-generator/issues/new/choose).
```

Comment:
These are the event types BYOB supports based on https://github.com/slsa-framework/github-actions-buildtypes/tree/main/workflow/v1

```suggestion
Only the following [event types] are supported:

| Supported event type  | Event description                          |
| --------------------- | ------------------------------------------ |
| [`create`]            | Creation of a git tag or branch.           |
| [`release`]           | Creation or update of a GitHub release.    |
| [`push`]              | Creation or update of a git tag or branch. |
| [`workflow_dispatch`] | Manual trigger of a workflow.              |

`pull_request` events are currently not supported. If you would like support for
`pull_request`, please tell us about your use case on
[issue #358](https://github.com/slsa-framework/slsa-github-generator/issues/358). If
you have an issue in all other triggers please submit a
[new issue](https://github.com/slsa-framework/slsa-github-generator/issues/new/choose).

[event types]: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows
[`create`]: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#create
[`release`]: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release
[`push`]: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#push
[`workflow_dispatch`]: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch
```